### ISSUE TITLE
Cell misalignment in server driven datagrid

### ISSUE THREAD

**Issue by runtothesun** [vmware/clarity#3635]
## Describe the bug
Row cells are not aligned with the column headings. Adjusting the column width or changing page will re render them correctly.

## How to reproduce
stackblitz reproduction:
https://stackblitz.com/edit/clarity-datagrid-fg1svu

See above

## Expected behavior
Row cells aligned with column headings

## Versions
* Angular: [8.1.2]
* Node: 
* Clarity: [2.0.2]

**Device:**

* Type: MacBook
* OS: MacOS
* Browser chrome, safari, IE11
* Version

## Additional notes

Was working prior to our upgrade to Angular 8 and Clarity v2

**Comment by runtothesun** [comment]
We also have another use case which has similar issues around column and data cell alignment since our update to Angular 8 and Clarity 2. In this case the column definitions are supplied with the data and this time cells appear to be randomly spaced (in a similar way to what happens if the number of <clr-dg-column> tags doesn't match the number of <clr-dg-cell> tags in each row). I've created another stackblitz to simulate this:  https://stackblitz.com/edit/clarity-datagrid-cell-spacing-issue

With the trackby function the columns render correctly when changing page, without it they don't. Adjusting the column widths doesn't cause it to render correctly unlike the previous issue.

If this is unrelated I'm happy to open another issue.

**Comment by gnomeontherun** [comment]
When I load your initial stackblitz, I only saw a problem on the initial load like the styles and application hadn't fully loaded. Then if I make any edit (and even then reset it) things load fine. This seems like a StackBlitz issue to me, can you see this happening in other places?

As far as the second issue example, I'm not sure exactly what is happening but our general recommendation is not to do this. There are a few reasons, 1) performance 2) dynamic calculations and 3) complexity. Not using ngFor means we don't have to recalculate (and drain performance) on recomputing all the datagrid sizes and that simplifies the API as well (its often less code to write the HTML elements out than all of the other code to build it dynamically).

**Comment by runtothesun** [comment]
I had upgraded to v2 and found the issue as shown in the second example so had to roll it back - then while I was building the example for the second issue I stumbled across the first, but unfortunately I didn't check that it occurred  in our actual application. You are right that it appears to be a Stackblitz problem as I can't recreate it outside of that environment.

I understand what you are saying about the second issue, it's just frustrating that it did work in v1! I'm sure we will be able to find a workaround. I'll close this now, thanks for your help.
